### PR TITLE
[Intel MKL] Fixing a failure in run_eager_op_as_function_test

### PR DESCRIPTION
**Pull request by mahmoud-abuzaina**
This PR fixes a failure in run_eager_op_as_function_test when MKL eager code is enabled. Failure started to happen after this [commit](https://github.com/tensorflow/tensorflow/commit/72e63b74a241c855c540997fd8f7cd2408c37016)

### CODE REVIEW

File: tensorflow/core/common_runtime/eager/execute.cc [hunk 1]
```diff
@@ -754,6 +755,16 @@ Status WrapInCallOp(EagerOperation* op, EagerOperation** wrapped_op) {
     for (const auto& attr : opdef.attr()) {
       (*ndef->mutable_attr())[attr.name()].set_placeholder(attr.name());
     }
+
+    if (IsMklEnabled() &&
+        absl::StartsWith(op->Name(), mkl_op_registry::kMklOpPrefix)) {
+      // All MKL eager ops have _kerenl private attribute that needs to be set
```
Comment (penpornk) [inline tensorflow/core/common_runtime/eager/execute.cc:761]: Nit:
```suggestion
      // All MKL eager ops have `_kernel` private attribute that needs to be set
```

**Review by penpornk — CHANGES_REQUESTED** [review]
Thank you for the fix!

File: tensorflow/core/common_runtime/eager/execute.cc (hunk 1 above)
Comment (mahmoud-abuzaina) [reply tensorflow/core/common_runtime/eager/execute.cc]: Fixed. Thanks.

**Review by penpornk — APPROVED** [review]
Thank you again for the PR!

**Review by gbaned — APPROVED** [review]

**Comment by penpornk** [comment]
@mahmoud-abuzaina Could you please help fix [Android Demo App](https://source.cloud.google.com/results/invocations/c0927a03-7bbb-4882-b277-f79cad512d2d) and [MacOS CPU Python 3](https://source.cloud.google.com/results/invocations/f3b83e7a-b03b-4970-8332-0b8d0e0b69de) errors? Thank you!
```
tensorflow/core/common_runtime/eager/execute.cc:760:38: error: use of undeclared identifier 'mkl_op_registry'
        absl::StartsWith(op->Name(), mkl_op_registry::kMklOpPrefix)) {
                                     ^
tensorflow/core/common_runtime/eager/execute.cc:764:25: error: use of undeclared identifier 'mkl_op_registry'
      attr_kernel.set_s(mkl_op_registry::kMklNameChangeOpLabel);
```

File: tensorflow/core/common_runtime/eager/execute.cc [hunk 2]
```diff
@@ -754,6 +755,16 @@ Status WrapInCallOp(EagerOperation* op, EagerOperation** wrapped_op) {
...
     }
+
+    if (IsMklEnabled() &&
+        absl::StartsWith(op->Name(), mkl_op_registry::kMklOpPrefix)) {
+      // All MKL eager ops have `_kernel` private attribute that needs to be set
+      // to a fixed label.
+      AttrValue attr_kernel;
+      attr_kernel.set_s(mkl_op_registry::kMklNameChangeOpLabel);
+      (*ndef->mutable_attr()).insert({"_kernel", attr_kernel});
+    }
```
Comment (penpornk) [inline tensorflow/core/common_runtime/eager/execute.cc:766]: Please wrap this code with `#ifdef INTEL_MKL` instead.

**Review by penpornk — CHANGES_REQUESTED** [review]
Thank you for the fix! I don't think we should always define `INTEL_MKL` though. Please add `if_mkl`s back.

File: tensorflow/core/common_runtime/eager/execute.cc (hunk 2 above)
Comment (mahmoud-abuzaina) [reply tensorflow/core/common_runtime/eager/execute.cc:767]: Fixed as suggested. Thank you!

**Review by penpornk — APPROVED** [review]
Thank you very much for the quick fix!

**Comment by mahmoud-abuzaina** [comment]
Looks like it is still failing. I think even the header file has to be under #ifdef INTEL_MKL. Let me do that.

**Review by penpornk — APPROVED** [review]
Thank you for the fix!

**Comment by penpornk** [comment]
Closing this as it is merged.

**Pull request closed without merging**